internal/jennies/java/converter.go

```diff
@@ -0,0 +1,167 @@
+package java
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/grafana/codejen"
+	"github.com/grafana/cog/internal/ast"
+	"github.com/grafana/cog/internal/jennies/common"
+	"github.com/grafana/cog/internal/jennies/template"
+	"github.com/grafana/cog/internal/languages"
+	"github.com/grafana/cog/internal/orderedmap"
+	"github.com/grafana/cog/internal/tools"
+)
+
+const variableModel = "VariableModel"
+
+type Converter struct {
+	config         Config
+	nullableConfig languages.NullableConfig
+	tmpl           *template.Template
+}
+
+func (jenny *Converter) JennyName() string {
+	return "JavaConverter"
+}
+
+func (jenny *Converter) Generate(context languages.Context) (codejen.Files, error) {
+	files := codejen.Files{}
+
+	for _, builder := range context.Builders {
+		output, err := jenny.generateConverter(context, builder)
+		if err != nil {
+			return nil, err
+		}
+
+		filename := filepath.Join(
+			jenny.config.ProjectPath,
+			formatPackageName(builder.Package),
+			fmt.Sprintf("%sConverter.java", tools.UpperCamelCase(builder.Name)),
+		)
+
+		files = append(files, *codejen.NewFile(filename, output, jenny))
+	}
+
+	var err error
+	for _, schema := range context.Schemas {
+		filterDataqueryObjects(schema).Iterate(func(key string, obj ast.Object) {
+			output, genErr := jenny.generateDataqueryConverter(context, schema, obj)
+			if genErr != nil {
+				err = genErr
+			} else {
+				filename := filepath.Join(
+					jenny.config.ProjectPath,
+					formatPackageName(schema.Package),
+					fmt.Sprintf("%sMapperConverter.java", tools.UpperCamelCase(obj.Name)),
+				)
+
+				files = append(files, *codejen.NewFile(filename, output, jenny))
+			}
+		})
+	}
+
+	return files, err
+}
+
+func filterDataqueryObjects(schema *ast.Schema) *orderedmap.Map[string, ast.Object] {
+	filteredObjects := orderedmap.New[string, ast.Object]()
+
+	schema.Objects.Iterate(func(key string, obj ast.Object) {
+		if obj.Type.ImplementedVariant() == string(ast.SchemaVariantDataQuery) {
+			if !obj.Type.HasHint(ast.HintSkipVariantPluginRegistration) {
+				filteredObjects.Set(key, obj)
+			}
+		}
+	})
+
+	return filteredObjects
```

Comment:
How about something like this instead?

```go
return schema.Objects.Filter(func(_ string, obj ast.Object) {
  return obj.IsDataqueryVariant() && !obj.Type.HasHint(ast.HintSkipVariantPluginRegistration)
})
```

Reply:
In the previous comment you said that that function affects to following jennies.

Reply:
That was because you were reassigning the result of `Filter()` to `schema.Objects`, which is not what I'm suggesting here :)